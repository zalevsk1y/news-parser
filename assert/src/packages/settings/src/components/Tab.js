import React from 'react';
import PropTypes from 'prop-types';
import Translate from '../containers/Translate'

class Tab extends React.Component {
    constructor(props){
      super(props);
      this.handleClick=this.handleClick.bind(this)
    }
    handleClick(){
      this.props.selectHandler(this.props.id)  
    }
    render(){
      const state=this.props.active,
            className="main-tabs "+this.props.className+(state?' active-tab':'');
      return (
        <h2 className={className} onClick={this.handleClick}><Translate>{this.props.name}</Translate></h2>
      );
    }
  }

Tab.propTypes={
  selectHandler:PropTypes.func.isRequired,
  className:PropTypes.string.isRequired,
  name:PropTypes.string.isRequired,
  id:PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ])

}

export default Tab; 