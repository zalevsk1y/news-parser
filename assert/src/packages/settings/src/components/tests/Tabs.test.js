import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {shallow, configure} from 'enzyme';

import Tab from '../Tab';
configure({adapter:new Adapter ()})

describe('Tab',()=>{
    const clickFn=jest.fn();
    const component=shallow(<Tab active={true} className='tab1' selectHandler={clickFn} id={1} name='testTab'/>);
    it('should render correctly ',()=>{
       
        expect(component).toMatchSnapshot()
    } )
    it('should call function passed to props selectHandler',()=>{
        component.find('h2').simulate('click',1);
        expect(clickFn).toHaveBeenCalledWith(1);
    })
})