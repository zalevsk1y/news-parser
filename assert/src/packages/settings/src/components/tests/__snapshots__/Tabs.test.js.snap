// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab should render correctly  1`] = `
<h2
  className="main-tabs tab1 active-tab"
  onClick={[Function]}
>
  testTab
</h2>
`;
